packages/opentelemetry-core/src/platform/node/sampling-probability.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ProbabilitySampler } from '../../trace/sampler/ProbabilitySampler';
+
+/**
+ * Probability is configured via a user set env var: OTEL_SAMPLING_PROBABILITY.
+ * Probability defaults to 1.
+ * @returns a probability sampler.
+ */
+export function getProbabilitySampler(): ProbabilitySampler {
+  return new ProbabilitySampler(
+    Number(process.env.OTEL_SAMPLING_PROBABILITY || 1)
```

Comment:
This is not a great way to convert a number to a string as there are many surprising edge cases

```typescript
const probability = typeof process.env.OTEL_SAMPLING_PROBABILITY === 'string' ? parseInt(process.env.OTEL_SAMPLING_PROBABILITY, 10) : 1;
if (isNaN(probability)) { return 1; } // parse can fail
return Math.max(0, Math.min(1, probabability)); // user can supply invalid number
```

Reply:
Thanks, one thing I don't get with the suggestion is:

```js
typeof process.env.OTEL_SAMPLING_PROBABILITY === 'string'`
```

Aren't env vars always strings? If so, would the following be fine?:

```js
const probability = parseInt(process.env.OTEL_SAMPLING_PROBABILITY, 10) ?? 1;
if (isNaN(probability)) { return 1; } // parse can fail
return Math.max(0, Math.min(1, probabability)); // user can supply invalid number
```

also ooc how could parse fail, if user provides something weird?

Reply:
After doing some reading it looks like `Number` doesn't have the same pitfalls that i'm used to with `parseInt`. In this case parsing can fail because the user can supply some value that isn't a number and we have to guard against that

```sh
$ node
> Number('this is not a number')
NaN
```

Reply:
I see, so we stick with `Number` but guard against `NaN` then?

Reply:
For invalid numbers I think `normalize` will take care of this:

https://github.com/open-telemetry/opentelemetry-js/blob/661b8cf121794c8149a33bd2ee406e7a52e85279/packages/opentelemetry-core/src/trace/sampler/ProbabilitySampler.ts#L45

Reply:
Should be fine to do it like this

```typescript
const probability = Number(process.env.OTEL_SAMPLING_PROBABILITY || 1);
if (isNaN(probability)) { return 1; } // parse can fail
return Math.max(0, Math.min(1, probabability)); // user can supply invalid number (too large or small)
```

Reply:
✅

Reply:
so in fact:

```js
> Number(process.env.FOFOFOF) || 1
1
> Number('fafafaf') || 1
1
```


just putting the 1 outside of the Number function is best